transform-plugins/src/main/java/co/cask/hydrator/plugin/Decompressor.java

```diff
@@ -0,0 +1,315 @@
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+    
+    for (Map.Entry<String, DecompressorType> entry : deCompMap.entrySet()) {
+      if (!outSchemaMap.containsKey(entry.getKey())) {
+        throw new IllegalArgumentException("Field '" + entry.getKey() + "' specified to be decompresed is not " +
+                                             "present in the output schema. Please add field '" + entry.getKey() + "'" +
+                                             "to output schema or remove it from decompress");
+      }
+      if (outSchemaMap.get(entry.getKey()) != Schema.Type.BYTES && 
+        outSchemaMap.get(entry.getKey()) != Schema.Type.STRING) {
+        throw new IllegalArgumentException("Field '" + entry.getKey() + "' should be of type BYTES or STRING in " +
+                                             "output schema.");
+      }
+    }
+  }
+  
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.decompressor);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+
+    Schema inSchema = in.getSchema();
+    List<Field> inFields = inSchema.getFields();
+
+    // Iterate through input fields. Check if field name is present
+    // in the fields that need to be decompressed, if it's not then write
+    // to output as it is.
+    for (Field field : inFields) {
+      String name = field.getName();
+
+      // Check if output schema also have the same field name. If it's not
+      // then continue.
+      if (!outSchemaMap.containsKey(name)) {
+        continue;
+      }
+
+      Schema.Type outFieldType = outSchemaMap. get(name);
+
+      // Check if the input field name is configured to be encoded. If the field is not
+      // present or is defined as none, then pass through the field as is.
+      if (!deCompMap.containsKey(name) || deCompMap.get(name) == DecompressorType.NONE) {
+        builder.set(name, in.get(name));
+      } else {
+        // Now, the input field should be of type byte[]
+        byte[] obj = new byte[0];
+        if (field.getSchema().getType() == Schema.Type.BYTES){
+          obj = in.get(name);
+        } else {
+          LOG.error("Input field '" + name + "' should be of type BYTES to decompress. It's currently of type" +
```

Comment:
should we continue executing after this condition? Otherwise should there be a ``continue``?